### PR TITLE
Hide DDG newtab of tor window in qwant region(ger/fr) and qwant is set

### PR DESCRIPTION
**Pull request by simonhong**
Hide DDG newtab of tor window in qwant region(ger/fr) if qwant is set to search provider.
Otherwise, DDG newetab is used in tor window newtab.

Close https://github.com/brave/brave-browser/issues/1632

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
1. Set region to france
2. Launch new browser instance with clean profile
3. Open tor window
4. Check search provider is qwant and qwant newtab is used
5. Set region to non fr/ger
6. Launch new browser instance with clean profile
7. Open tor window
8. Check search provider is ddg and ddg newtab is used

## Reviewer Checklist:

- [x] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [x] Adequate test coverage exists to prevent regressions 
- [x] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by simonhong** [comment]
Marked as WIP because after changing search provider and then reloading newtab doesn't reflect new search provider. But, when relaunch new search provider configuration is reflected in tor newtab.

**Comment by simonhong** [comment]
I think this is ready to review!

File: browser/ui/webui/brave_new_tab_ui.cc [hunk 1]
```diff
@@ -41,6 +44,17 @@ class NewTabDOMHandler : public content::WebUIMessageHandler {
...
 };
 
+bool IsQwantUsedInQwantRegion(Profile* profile) {
+  bool region_for_qwant =
+      TemplateURLPrepopulateData::GetPrepopulatedDefaultSearch(
+          profile->GetPrefs())->prepopulate_id ==
+          TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_QWANT;
+  bool qwant_used =
+      profile->GetPrefs()->GetInteger(kAlternativeSearchEngineProviderInTor) ==
+      TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_QWANT;
```
Comment (bsclifton) [inline browser/ui/webui/brave_new_tab_ui.cc]: `qwant_used` only returns true for Tor tabs with this logic; this should return true also for incognito windows

**Comment by bsclifton** [comment]
master 0bd21c10b387c2c3dd2bcde1929dce12ebc28e1e
0.57.x 2e582a9afd4ee5d0cf4b0f39370225b20a3422cb
0.56.x 1b6d35372bf9cca7cd692f0b71f761b25889c6a8
0.55.x 76d2c1d9dced19f75b3c29170ee26f5e3fae1989

**Pull request merged**